### PR TITLE
separate SVG for legend

### PR DESCRIPTION
**Pull request by tingxuanz**
@rowlandm , use a separate SVG for the legend. The legend is now scrollable

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -345,19 +345,19 @@ setup_brush = function(graph){
           .attr("transform", function(d, i) { return "translate(0,"+ i * 20 +")"; });
 
       legend.append("rect")
-          .attr("x", width + 20)
```
Comment (rowlandm) [inline lib/index.js:348]: Why are we going from width + 20 to an integer (1)

Shouldn't there by a length you pass in?

Comment (tingxuanz) [reply lib/index.js:348]: In the old version, the pca plot and the legend are in the same svg. "width" is the width for pca plot. The legend is located at the position (width+20).

Comment (rowlandm) [reply lib/index.js:348]: But the width for the legend should be passed in ? I don't think we want to hardcode it.

Comment (tingxuanz) [reply lib/index.js:348]: In the new commit, the width for the legend is passed in as legend_width, a property of options.

Comment (rowlandm) [reply lib/index.js:348]: OK. Sounds good.

Comment (rowlandm) [reply lib/index.js:348]: Feel free to merge.

**Pull request merged**